packages/core/core/src/utils.js

```diff
@@ -51,3 +52,35 @@ export function registerCoreWithSerializer() {
 
   coreRegistered = true;
 }
+
+export function getPublicId(
+  id: string,
+  alreadyExists: string => boolean,
+): string {
+  assert(
+    id.match(/^[0-9a-f]{32}$/),
+    `id ${id} must be a 32-character hexadecimal string`,
+  );
+
+  let buf = Buffer.alloc(16);
+  for (let byteOffset = 0; byteOffset < 16; byteOffset += 4) {
+    // Add the integer values to the buffer
+    // https://stackoverflow.com/questions/8044543/how-can-i-store-an-integer-in-a-nodejs-buffer/53550757#53550757
+    buf.writeUInt32BE(
+      parseInt(id.slice(byteOffset * 2, byteOffset * 2 + 8), 16),
+      byteOffset,
+    );
+  }
+
+  // 128-bit values are represented in the first 22 characters
+  // of the base64 string. The remaining two are always '=='.
+  let base64 = buf.toString('base64').slice(0, 22);
```

Comment:
Maybe good to remove special characters like we did in Parcel 1 instead (as we never really use the entire string anyway), there's also other characters in here besides `=`

Reply:
All base64 characters should be fine in a string literal where they're currently used. Did you have another use in mind?

Reply:
It was just a nitpick in case it would ever be used as part of a variable name or something, like we did with scope-hoisting, but that still uses the long id's. You can leave it as is